src/ray/core_worker/test/normal_task_submitter_test.cc

```diff
@@ -1863,19 +1904,20 @@ TEST(NormalTaskSubmitterTest, TestBacklogReport) {
   TaskSpecification task3 = BuildTaskSpec(resources1, descriptor1);
   task3.GetMutableMessage().add_args()->mutable_object_ref()->set_object_id(
       plasma2.Binary());
-  TestSchedulingKey(store, task2, task2, task3);
+  TestSchedulingKey(store, WithRandomTaskId(task2), WithRandomTaskId(task2), task3);
 
   TaskSpecification task4 = BuildTaskSpec(resources2, descriptor2);
 
   ASSERT_TRUE(submitter.SubmitTask(task1).ok());
   // One is requested and one is in the backlog for each SchedulingKey
-  ASSERT_TRUE(submitter.SubmitTask(task2).ok());
-  ASSERT_TRUE(submitter.SubmitTask(task2).ok());
-  ASSERT_TRUE(submitter.SubmitTask(task3).ok());
-  ASSERT_TRUE(submitter.SubmitTask(task3).ok());
-  ASSERT_TRUE(submitter.SubmitTask(task4).ok());
-  ASSERT_TRUE(submitter.SubmitTask(task4).ok());
-
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task2)).ok());
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task2)).ok());
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task3)).ok());
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task3)).ok());
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task4)).ok());
+  ASSERT_TRUE(submitter.SubmitTask(WithRandomTaskId(task4)).ok());
+
+  std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
why sleep?

Reply:
We need to wait for submitter.SubmitTask()'s callback to finish. Previously it's finished sync, now it's async.

Reply:
Can we do something like io_context.run to execute the callback instead of relying on a random 1 seconds?

Reply:
tried and it works. Thanks for the suggestion!